plugins/bulk-import/src/components/AddRepositories/RepositoriesTable.tsx

```diff
@@ -0,0 +1,258 @@
+import * as React from 'react';
+
+import { makeStyles } from '@material-ui/core';
+import Table from '@mui/material/Table';
+import TableBody from '@mui/material/TableBody';
+import TableCell from '@mui/material/TableCell';
+import TableContainer from '@mui/material/TableContainer';
+import TablePagination from '@mui/material/TablePagination';
+import TableRow from '@mui/material/TableRow';
+import { FormikErrors } from 'formik';
+
+import {
+  AddRepositoriesData,
+  AddRepositoriesFormValues,
+  Order,
+} from '../../types';
+import {
+  getComparator,
+  getNewSelectedRepositories,
+} from '../../utils/repository-utils';
+import { getDataForRepositories } from './mockData';
+import { OrganizationTableRow } from './OrganizationTableRow';
+import { RepositoriesColumnHeader } from './RepositoriesColumnHeader';
+import { RepositoriesHeader } from './RepositoriesHeader';
+import { RepositoryTableRow } from './RepositoryTableRow';
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    alignItems: 'start',
+    padding: theme.spacing(3, 0, 2.5, 2.5),
+  },
+  empty: {
+    padding: theme.spacing(2),
+    display: 'flex',
+    justifyContent: 'center',
+  },
+  title: {
+    display: 'flex',
+    gap: '20px',
+    alignItems: 'center',
+  },
+  footer: {
+    '&:nth-of-type(odd)': {
+      backgroundColor: `${theme.palette.background.paper}`,
+    },
+  },
+}));
+
+export const RepositoriesTable = ({
+  searchString,
+  selectedRepositoriesFormData,
+  page,
+  setPage,
+  setFieldValue,
+  showOrganizations = false,
+}: {
+  searchString: string;
+  selectedRepositoriesFormData: AddRepositoriesFormValues;
+  page: number;
+  setPage: (page: number) => void;
+  setFieldValue: (
+    field: string,
+    value: any,
+    shouldValidate?: boolean,
+  ) => Promise<FormikErrors<AddRepositoriesFormValues>> | Promise<void>;
+  showOrganizations?: boolean;
+}) => {
+  const classes = useStyles();
+  const [order, setOrder] = React.useState<Order>('asc');
+  const [orderBy, setOrderBy] = React.useState<string>('name');
+  const [selected, setSelected] = React.useState<number[]>([]);
+
+  const [rowsPerPage, setRowsPerPage] = React.useState(5);
+
+  const data: AddRepositoriesData[] = getDataForRepositories();
+
+  const handleRequestSort = (
+    _event: React.MouseEvent<unknown>,
+    property: string,
+  ) => {
+    const isAsc = orderBy === property && order === 'asc';
+    setOrder(isAsc ? 'desc' : 'asc');
+    setOrderBy(property);
+  };
+
+  const handleSelectAllClick = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (event.target.checked) {
+      const newSelected = data
```

Comment:
Just a question, if user filtered data then click select all checkbox, will it select all data or only filtered data?

Reply:
Fixed it


https://github.com/janus-idp/backstage-plugins/assets/22490998/321443e5-960b-4839-bf0b-e8217cee63b4